### DISCUSSION TITLE
Apply PHP-CS-Fixer only on diffed lines in a given commit

### DISCUSSION THREAD

**Discussion by tomcodes** [PHP-CS-Fixer/PHP-CS-Fixer#7853] [Q&A]
Hello!

I am trying to introduce PHP-CS-Fixer in a code base where more developers are joining. It seems we do not have the same cult-like belief in PSR rules so I want to enforce a universal code-style across the whole project.

I am currently reviewing a substantial PR (90 changed files, + 2097, − 1273) and most of the new / updated code needs CS changes.

Is there a way to have PHP-CS-Fixer apply not on whole files (which would make the code review process nonsense) but only on diffed lines in a given commit?

Thank you.

**Comment by Wirone** [comment]
Please take a look [here](https://cs.symfony.com/doc/usage.html#using-php-cs-fixer-on-ci). In general, you need to determine the changed files' list on your own and pass them as arguments of the `check` / `fix` command.

It's also possible to do it on Git hooks before commit/push, e.g. with [Captain Hook](https://github.com/captainhookphp/captainhook).
